internal/expr/query.go

```diff
@@ -174,7 +181,18 @@ func (qe *QueryExpr) ScanArgs(columns []string, outputArgs []any) (scanArgs []an
 			if !val.CanSet() {
 				return nil, nil, fmt.Errorf("internal error: cannot set field %s of struct %s", tm.name, tm.structType.Name())
 			}
-			ptrs = append(ptrs, val.Addr().Interface())
+			pt := reflect.PointerTo(val.Type())
+			if val.Type().Kind() != reflect.Pointer && !pt.Implements(scannerInterface) {
+				// Rows.Scan will return an error if it tries to scan NULL into a type that does not have nil.
+				// Pointers are the only nil type that work with Rows.Scan so for none pointer types we generate
+				// a pointer to them and pass this to Rows.Scan. After scanning the variable is set, if the
```

Comment:
This sentence doesn't seem to be reading properly. Needs some rewording.

Reply:
Changed as discussed in meeting.